resource_obmcs_core_instance_test.go

```diff
@@ -109,6 +117,91 @@ func (s *ResourceCoreInstanceTestSuite) TestAccResourceCoreInstance_basic() {
 				}`,
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "-tf-instance"),
+					func(ts *terraform.State) (err error) {
+						newId, err := fromInstanceState(ts, s.ResourceName, "id")
+						if newId != instanceId {
+							return fmt.Errorf("Expected same instance ocid, got the different.")
+						}
+						return err
+					},
+				),
+			},
+			// verify force refresh by creating an instance with create_vnic_details.
+			{
+				Config: s.Config + `
+				resource "oci_core_instance" "t" {
+					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
+					compartment_id = "${var.compartment_id}"
+					image = "${data.oci_core_images.t.images.0.id}"
+					shape = "VM.Standard1.1"
+					display_name = "-tf-instance"
+					metadata {
+						ssh_authorized_keys = "${var.ssh_public_key}"
+					}
+					create_vnic_details {
+							subnet_id = "${oci_core_subnet.t.id}"
+							display_name = "-tf-vnic"
+					}
+				}
+				data "oci_core_vnic_attachments" "t" {
+					compartment_id = "${var.compartment_id}"
+					instance_id = "${oci_core_instance.t.id}"
+				}
+				data "oci_core_vnic" "t" {
+					vnic_id = "${lookup(data.oci_core_vnic_attachments.t.vnic_attachments[0],"vnic_id")}"
+				}`,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(s.ResourceName, "display_name", "-tf-instance"),
+					resource.TestCheckResourceAttr(vnicResourceName, "display_name", "-tf-vnic"),
+					resource.TestCheckResourceAttr(vnicResourceName, "skip_source_dest_check", "false"),
+					func(ts *terraform.State) (err error) {
+						newId, err := fromInstanceState(ts, s.ResourceName, "id")
+						if newId == instanceId {
+							return fmt.Errorf("Expected new instance ocid, got the same.")
+						}
+						instanceId = newId
+						return err
+					},
+				),
+			},
+			// verify force refresh by creating an instance with different VNIC details.
+			{
+				Config: s.Config + `
+				resource "oci_core_instance" "t" {
+					availability_domain = "${data.oci_identity_availability_domains.ADs.availability_domains.0.name}"
+					compartment_id = "${var.compartment_id}"
+					image = "${data.oci_core_images.t.images.0.id}"
+					shape = "VM.Standard1.1"
+					display_name = "-tf-instance"
+					metadata {
+						ssh_authorized_keys = "${var.ssh_public_key}"
+					}
+					create_vnic_details {
+							subnet_id = "${oci_core_subnet.t.id}"
+							display_name = "-tf-vnic"
+							assign_public_ip = true
+							private_ip = "10.0.1.20"
+							skip_source_dest_check = true
```

Comment:
extra tabs, be sure to run "go fmt" on your changed files prior to commit

Reply:
fixed - go fmt doesn't actually catch this, I assume since it's inside of a string.